Add post about WildFly Docker and new provisioning architecture

jmesnil opened the pull request:
Signed-off-by: Jeff Mesnil <jmesnil@redhat.com>

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+---
+layout: post
+title: 'Use the wildfly-maven-plugin to create Docker image of your application'
```
s/create/create a/g

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+author: jmesnil
+---
+
+In this article, we will explain how a developer using the Docker image for WildFly can take advantage of the new capabilities of the `wildlfy-maven-plugin` and the new WildFy runtime image to build their container image.
```
s/WildFy/WildFly

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+These two types of images, the "vanilla" Docker image and the OpenShift S2I images, have no connection and no synergy.
+A lot of work has been done in WildFly around provisioning. The S2I images benefitted from this effort but the Docker image did not... until now.
+
+While S2I is a good technology to build application image on OpenShift, we realized that we could achieve a more flexible solution that would benefit other container platforms (Kubernetes, Azure) as well as users running WildFly on premise.
```
s/image/images or s/application/an application/g

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+
+While S2I is a good technology to build application image on OpenShift, we realized that we could achieve a more flexible solution that would benefit other container platforms (Kubernetes, Azure) as well as users running WildFly on premise.
+
+We decided to focus on a Maven-centric approach to provision WildFly so that Maven (and the user) would be in control of the complete runtime (including the user deployment as well as WildFly runtime).
```
s/as well as WildFly/as well as the WildFly/g

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+
+## WildFly Docker image
+
+There are different ways to use the https://quay.io/repository/wildfly/wildfly?tab=info[quay.io/wildfly/wildfly] image to create an *application image* (that would contain both WildFly and your deployments) but the simplest one is to add the deployment archive to this base image and let WildFly deploys it when it starts.
```
s/deploys/deploy

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+Let's now move to the new architecture for WildFly.
+Jean-Francois Denise extensively described it in the https://www.wildfly.org/news/2022/04/20/WildFly-s2i-v2-Released/[WildFly S2I new architecture is final!] article and focused on the Source-to-Image (S2I) capabilities of the architecture.
+
+In this article, we will see that this architecture also benefits users that are not using S2I and wants to keep control of the creation of their application images.
```
s/users that/users who/g
s/wants to keep/want to keep/g

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+
+## Moving from WildFly Docker image to Runtime image
+
+Let's resume what is needed to move from the vanilla Docker image to the new runtime image for WildFly:
```
s/resume/review

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+* It reduces the security risk as you are not subject to security attacks if the affected components are not present in application at all.
+* It reduces the size of the server runtime.
+
+In our example, our `microprofile-config` quickstart requires MicroProfile to run. WildFly provides a convenient `microprofile-platform` that provision everything that is needed to run MicroProfile applications. We can trim our runtime with only this layer by updating the `wildfly-maven-plugin`:
```
s/provision/provisions

bstansberry replied:
s/with only/to only/g

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+
+## Packaging Scripts
+
+The `wildfly-maven-plugin` also provides the ability to execute JBoss CLI commands when WildFly is provisioned. This allows to substantially modify the standalone configuration to better fit the application requirements. It has no impact on the application image as these scripts are only invoked during provisioning. 
```
s/allows to/allows you to/g

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+## Feature Packs
+
+WildFly uses feature packs as the building blocks to provision the server.
+The most important feature pack is WildFly's own feature pack: `org.wildfly:wildfly-galleon-pack:26.1.1.Final` to contols the installation of WildFly itself.
```
s/controls/control

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+We are also providing additional feature packs to provide additional capabilities to WildFly.
+It is out of scope of this article to list all of them but let's discuss two interesting ones:
+
+* The https://github.com/wildfly-extras/wildfly-cloud-galleon-pack[wildfly-cloud-galleon-pack] provides a set of additional features allowing you to configure a WildFly server to work on the cloud. It adapts WildFly to run on orchestration plaftorms in an optimized way. In particular, it automatically route server logs to the console, it provisions the `health` subsystem to monitor the server hea ...
```
s/route/routes
s/healthiness/health

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+## Conclusion
+
+The `wildfly-maven-plugin` is currently at version 4.0.0.Beta2 with a Final release planned for WildFly 27.
+It builds on top of the experience we gained from the Bootable Jar and provide a compelling architecture to control the full runtime (WildFly + the application deployments) from the application's `pom.xml`.
```
s/provide/provides

bstansberry commented on _posts/2022-08-05-wildfly-maven-docker.adoc:
```diff
@@ -0,0 +1,272 @@
+
+The `wildfly-maven-plugin` is currently at version 4.0.0.Beta2 with a Final release planned for WildFly 27.
+It builds on top of the experience we gained from the Bootable Jar and provide a compelling architecture to control the full runtime (WildFly + the application deployments) from the application's `pom.xml`.
+The full customization of WildFly (using feature packs, packaging scripts, etc.) is controlled by the developer so that the runtime fits the uer's application.
```
s/uer's/user's/g

bstansberry reviewed (changes requested):
General content looks good; I've just made some copy-editing notes.

jmesnil commented:
thanks @bstansberry!

I open this PR as draft as it was not ready for review and needed more editing. Your review confirmed that! :) 
If you are ok with the structure and content, I'll polish it a bit more and we should be good

jmesnil commented:
@bstansberry I've addressed all your requests and added https://github.com/wildfly/wildfly.org/compare/54c0b8fc25d9a443feb5e82f9b8ec33687c03acd..b3821e39ca55cd872c376702507249e81dd8d430
I think we can rationalise our effort around our images (Docker, S2I, maybe buildpacks in a future) and wants to start a conversation on wildfly-dev about that. I see this article as a way to aggregate feedbacks on this effort.
